packages/react/src/components/Tabs/tabs.test.tsx

```diff
@@ -0,0 +1,411 @@
+          <Tab target={ref2}>option 2</Tab>
+          <Tab target={ref3}>option 3</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+        <TabPanel ref={ref2}>
+          <p>Panel 2</p>
+        </TabPanel>
+        <TabPanel ref={ref3}>
+          <p>Panel 3</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  fireEvent.click(screen.getAllByRole('tab')[0]);
+  await waitFor(() => {
+    expect(screen.getAllByRole('tabpanel')[0]).not.toHaveClass(
+      'TabPanel--hidden'
+    );
+    expect(screen.getAllByRole('tabpanel')[1]).toHaveClass('TabPanel--hidden');
+    expect(screen.getAllByRole('tabpanel')[2]).toHaveClass('TabPanel--hidden');
+  });
+
+  fireEvent.click(screen.getAllByRole('tab')[1]);
+  await waitFor(() => {
+    expect(screen.getAllByRole('tabpanel')[0]).toHaveClass('TabPanel--hidden');
+    expect(screen.getAllByRole('tabpanel')[1]).not.toHaveClass(
+      'TabPanel--hidden'
+    );
+    expect(screen.getAllByRole('tabpanel')[2]).toHaveClass('TabPanel--hidden');
+  });
+});
+
+test('should focus the active tab with keyboard activation', async () => {
+  const mountElement = document.createElement('div');
+  document.body.appendChild(mountElement);
+  const TabsComponent = () => {
+    const ref = useRef(null);
+    const ref2 = useRef(null);
+
+    return (
+      <>
+        <Tabs aria-label={ariaLabel}>
+          <Tab id="tab-uno" target={ref}>
+            option 1
+          </Tab>
+          <Tab id="tab-dos" target={ref2}>
+            option 2
+          </Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+        <TabPanel ref={ref2}>
+          <p>Panel 2</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabsComponent />, { container: mountElement });
+  fireEvent.keyDown(screen.getByText('option 2'), { key: 'ArrowRight' });
+  await waitFor(() => {
+    expect(screen.getByText('option 2')).toHaveClass('Tab--active');
+  });
+});
+
+test('should display correct TabPanel when clicking a tab with a customized id', async () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    const ref2 = useRef(null);
+
+    return (
+      <>
+        <Tabs aria-label={ariaLabel} id="customized-id">
+          <Tab target={ref}>option 1</Tab>
+          <Tab target={ref2}>option 2</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+        <TabPanel ref={ref2}>
+          <p>Panel 2</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  fireEvent.click(screen.getAllByRole('tab')[1]);
+  await waitFor(() => {
+    expect(screen.getAllByRole('tabpanel')[0]).toHaveClass('TabPanel--hidden');
+    expect(screen.getAllByRole('tabpanel')[1]).not.toHaveClass(
+      'TabPanel--hidden'
+    );
+  });
+});
+
+test('should display correct tabpanel when pressing left, right, home, or end keys', async () => {
```

Comment:
This doesn't actually test the left keydown event. I'm guessing that's because it's covered in the test on line 179, maybe we should just combine them?